Documentation/platforms/aws.md

```diff
@@ -0,0 +1,77 @@
+
+* distribution of EC2 instances across Availability Zones (AZs)
+* EC2 instance fail over across availability zones
+* consolidated monitoring and life cycle control of instances within an ASG
+
+The use of an ASG to create a [self healing etcd cluster](#self-healing) is one of the design considerations when deploying an etcd cluster to AWS.
+
+## Cluster Design
+
+The purpose of this section is to provide foundational guidance for deploying etcd on AWS. The discussion will be framed by the following three critical design criteria about the etcd cluster itself:
+
+* block device provider: limited to the tradeoffs between EBS or instance storage (InstanceStore)
+* cluster topology: how many nodes should make up an etcd cluster; should these nodes be distributed over multiple AZs
+* managing etcd members: creating a static cluster of EC2 instances or using an ASG.
+
+The intended cluster workload will is the primary driver of cluster design decisions. A configuration store for microservices may require different design considerations than a distributed lock service, a secrets store, or the Kubernetes control plane. Cluster design tradeoffs include considerations such as
+
+* availability
+* data safety
+* performance/throughput
+* self healing
+
+which must be matched to the expectations of the intended workload to be served by the etcd cluster.
+
+### Availability
+
+Instance availability on AWS is ultimately determined by the Amazon EC2 Region Service Level Agreement (SLA) which is the policy by which Amazon describes their precise definition of a regional outage.
+
+> “Region Unavailable” and “Region Unavailability” mean that more than one Availability Zone in which you are
+running an instance, within the same Region, is “Unavailable” to you.
+
+In the context of an etcd cluster this means you must have at minimum a three member cluster where EC2 instances are spread across at least two AZs in order for an etcd cluster to be considered highly available at a Regional level.
+
+For most use cases the additional latency associated with a cluster spanning across Availability Zones will introduce a negligible performance impact.
+
+Availability considerations apply to all components of an application; if the application which accesses the etcd cluster will only be deployed to a single Availability Zone it may not make sense to make the etcd cluster highly available across zones.
+
+### Data Safety
+
+There are multiple mechanisms by which data safety can be achieved when deploying etcd on AWS.
+
+* replication: etcd replicates all data to all members of the etcd cluster therefore more nodes and disks you have in your cluster and the stronger the isolation between failure domains, the smaller the probability that data persisted to an etcd cluster will be unavailable to an application
```

Comment:
* replication: etcd replicates all its data to all its members. Therefore, given more members in the cluster and more independent failure domains, the less likely an etcd cluster will be permanently lost.